relay-dynamic-config/src/metrics.rs

```diff
@@ -124,3 +125,178 @@ impl TransactionMetricsConfig {
+    ///
+    /// If the field does not exist, extraction is skipped.
+    #[serde(default)]
+    pub field: Option<String>,
+
+    /// An optional condition to apply before extraction.
+    ///
+    /// See [`RuleCondition`] for all available options to specify and combine conditions. If no
+    /// condition is specified, the metric is extracted unconditionally.
+    #[serde(default)]
+    pub condition: Option<RuleCondition>,
+
+    /// A list of tags to add to the metric.
+    ///
+    /// Tags can be conditional, see [`TagSpec`] for configuration options. For this reason, it is
+    /// possible to list tag keys multiple times, each with different conditions. The first matching
+    /// condition will be applied.
+    #[serde(default)]
+    pub tags: Vec<TagSpec>,
+}
+
+/// Mapping between extracted metrics and additional tags to extract.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TagMapping {
+    /// A list of metrics identifiers (MRI) to apply tags to.
+    ///
+    /// Entries in this list can contain wildcards to match metrics with dynamic MRIs.
+    #[serde(default)]
+    pub metrics: Vec<String>,
+
+    /// A list of tags to add to the metric.
+    ///
+    /// Tags can be conditional, see [`TagSpec`] for configuration options. For this reason, it is
+    /// possible to list tag keys multiple times, each with different conditions. The first matching
+    /// condition will be applied.
+    #[serde(default)]
+    pub tags: Vec<TagSpec>,
+}
+
+/// Configuration for a tag to add to a metric.
+///
+/// Tags values can be static if defined through `value` or dynamically queried from the payload if
+/// defined through `field`. These two options are mutually exclusive, behavior is undefined if both
+/// are specified.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TagSpec {
+    /// The key of the tag to extract.
+    pub key: String,
+
+    /// Path to a field containing the tag's value.
+    ///
+    /// Mutually exclusive from `value`.
+    #[serde(default)]
+    pub field: Option<String>,
+
+    /// Literal value of the tag.
+    ///
+    /// Mutually exclusive from `field`.
+    #[serde(default)]
+    pub value: Option<String>,
+
+    /// An optional condition to apply before extraction.
+    ///
+    /// See [`RuleCondition`] for all available options to specify and combine conditions. If no
+    /// condition is specified, the tag is added unconditionally, provided it is not already there.
+    #[serde(default)]
+    pub condition: Option<RuleCondition>,
+}
+
+impl TagSpec {
+    /// Returns the source of tag values, either literal or a field.
+    pub fn source(&self) -> TagSource<'_> {
+        if let Some(ref field) = self.field {
+            TagSource::Field(field)
+        } else if let Some(ref value) = self.value {
+            TagSource::Literal(value)
+        } else {
+            TagSource::Unknown
```

Comment:
What do you think about logging an error in this case, at least for the first few iterations?

Reply:
If we make changes during our first iterations, this error would also show up for customers if they run a Relay with this intermediate version. Since this is future-proofing, it's better if we keep this silent. There can be other tests that ensure our config roundtrips in Sentry.